modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ConcurrentLaws.kt

```diff
@@ -670,7 +670,7 @@ object ConcurrentLaws {
         val errorLatch = Promise<F, Int>(this@onErrorIsRunWhenErrorIsRaised).bind()
 
         startLatch.complete(Unit).flatMap { raiseError<Exception>(RuntimeException("Boom")) }
-          .onError(errorLatch.complete(i))
+          .onError { errorLatch.complete(i) }
```

Comment:
```suggestion
          .onError(errorLatch::complete)
```

This way you can use `Gen.throwable()` and assert that the correct pas used `!effect { waitExit shouldBe generatedThrowable }` and verify with `Eq<Kind<F, Unit>>` instead.